query/src/main/java/org/infinispan/query/clustered/AbstractQueryDefinitionExternalizer.java

```diff
@@ -0,0 +1,71 @@
+package org.infinispan.query.clustered;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import org.hibernate.search.query.engine.spi.HSQuery;
+import org.infinispan.commons.marshall.AdvancedExternalizer;
+import org.infinispan.query.QueryDefinition;
+
+/**
+ * @since 9.2
+ */
+public abstract class AbstractQueryDefinitionExternalizer<Q extends QueryDefinition> implements AdvancedExternalizer<Q> {
+
+   @Override
+   public void writeObject(ObjectOutput output, Q object) throws IOException {
+      if (object.getQueryString().isPresent()) {
+         output.writeBoolean(true);
+         output.writeUTF(object.getQueryString().get());
+      } else {
+         output.writeBoolean(false);
+         output.writeObject(object.getHsQuery(null));
```

Comment:
I don't understand how this can work with ```null``` cache parameter.

Reply:
ops, let me clean this up